sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/TestingServiceVersions.java

```diff
@@ -0,0 +1,83 @@
+     * <p>
+     * This is used to convert {@link #minimumLiveServiceVersion()}, {@link #maximumLiveServiceVersion()}, and {@link
+     * #recordingServiceVersion()} into their {@link ServiceVersion} type.
+     * <p>
+     * If neither of {@link #minimumLiveServiceVersion()} and {@link #maximumLiveServiceVersion()} are provided all
+     * {@link ServiceVersion ServiceVersions} represented by this type will be used. If {@link
+     * #recordingServiceVersion()} isn't configured the latest {@link ServiceVersion} represented by this type will be
+     * used.
+     * <p>
+     * If this type isn't an {@link Enum} an {@link IllegalArgumentException} will be thrown during runtime.
+     *
+     * @return The class that represents the service version type.
+     */
+    Class<? extends ServiceVersion> serviceVersionType();
+
+    /**
+     * The minimum service version used during {@link TestMode#LIVE} testing.
+     * <p>
+     * If neither this or {@link #maximumLiveServiceVersion()} are set all service versions will be used. If only this
+     * is set then all service versions equal to or later than the set version will be used. If both this and {@link
+     * #maximumLiveServiceVersion()} are set then the inclusive range of service versions will be used.
+     * <p>
+     * If no service versions meet the requirements of this and {@link #maximumLiveServiceVersion()} an {@link
+     * IllegalStateException} will be thrown during runtime.
+     * <p>
+     * {@link ServiceVersion ServiceVersions} are compared using their enum ordinal.
+     *
+     * @return The minimum service version that will be used.
+     */
+    String minimumLiveServiceVersion() default "";
+
+    /**
+     * The maximum service version used during {@link TestMode#LIVE} testing.
+     * <p>
+     * If neither this or {@link #minimumLiveServiceVersion()} are set all service versions will be used. If only this
+     * is set then all service versions less than or equal to the set version will be used. If both this and {@link
+     * #minimumLiveServiceVersion()} are set then the inclusive range of service versions will be used.
+     * <p>
+     * If no service versions meet the requirements of this and {@link #minimumLiveServiceVersion()} an {@link
+     * IllegalStateException} will be thrown during runtime.
+     * <p>
+     * {@link ServiceVersion ServiceVersions} are compared using their enum ordinal.
+     *
+     * @return The minimum service version that will be used.
+     */
+    String maximumLiveServiceVersion() default "";
+
+    /**
+     * The service version used when running in {@link TestMode#RECORD} and {@link TestMode#PLAYBACK} mode.
+     * <p>
+     * If this isn't set the latest service version of {@link #serviceVersionType()} is used.
+     *
+     * @return The service version used when running in {@link TestMode#RECORD} and {@link TestMode#PLAYBACK} mode.
+     */
+    String recordingServiceVersion() default "";
```

Comment:
Why do we need separate versions for LIVE and RECORD modes? If we run tests against all versions in LIVE mode, we should be able to record them as well.

Reply:
Being removed